swc/private/swc_plugin.bzl

```diff
@@ -0,0 +1,33 @@
+"Implementation details for swc_plugin rule"
+
+load("//swc:providers.bzl", "SwcPluginConfigInfo")
+
+_attrs = {
+    "src": attr.label(
+        doc = "label for the plugin, either a directory containing a package.json pointing at a wasm file as the main entrypoint, or a wasm file",
+        providers = [DefaultInfo],
+        mandatory = True,
+        allow_files = True,
+    ),
+    "config": attr.string(
+        doc = "configuration object for the plugin, serialized JSON",
```

Comment:
why not have the rule accept a dictionary here too, just like the macro? we can `json.encode` in the rule implementation just as easily

Reply:
I think this has to be a string, `string_dict` wouldn't be enough as the plugin config can be an arbitrary JSON object.